x-pack/qa/oidc-op-tests/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtWithOidcAuthIT.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authc.jwt;
+
+import com.nimbusds.oauth2.sdk.ResponseType;
+import com.nimbusds.oauth2.sdk.Scope;
+import com.nimbusds.oauth2.sdk.id.ClientID;
+import com.nimbusds.oauth2.sdk.id.State;
+import com.nimbusds.openid.connect.sdk.AuthenticationRequest;
+import com.nimbusds.openid.connect.sdk.Nonce;
+import com.nimbusds.openid.connect.sdk.OIDCScopeValue;
+
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.rest.RestUtils;
+import org.elasticsearch.test.TestMatchers;
+import org.elasticsearch.test.TestSecurityClient;
+import org.elasticsearch.xpack.core.security.user.User;
+import org.elasticsearch.xpack.security.authc.oidc.C2IdOpTestCase;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.hasKey;
+
+/**
+ * Tests that the Elasticsearch "jwt" realm ({@link JwtRealm}) can be used to authenticate using a JWT that is actially an ID Token
+ * from a real OIDC Originating Party (OP).
+ * This test is intentionally minimal - the bulk of the JWT functionality is tested elsewhere - the purpose of this test is purely to
+ * verify successful integration an OIDC OP.
+ */
+public class JwtWithOidcAuthIT extends C2IdOpTestCase {
+
+    // configured in the Elasticearch node test fixture
+    private static final String C2ID_AUTH_ENDPOINT = "http://oidc-provider:8080/c2id-login";
+    private static final List<String> ALLOWED_AUDIENCES = List.of("elasticsearch-jwt1", "elasticsearch-jwt2");
+    private static final String JWT_REALM_NAME = "op-jwt";
+
+    // Constants for role mapping
+    private static final String ROLE_NAME = "jwt_role";
+
+    // Randomised values
+    private static String clientId;
+    private static String redirectUri;
+
+    @BeforeClass
+    public static void registerClients() throws IOException {
```

Comment:
Alternatively, does it make sense to consolidate them and move to the C2IdOpTestCase parent class. If client provisioning in the two test classes overlap, then it might not make sense, unless you can calls to clean up in an @AfterClass method.

Reply:
> My assumption is all OIDC OPs register clients statically, not dynamically.

It's actually the reverse. Almost all OIDC OPs register dynamically. SAML is more likely to be a mix of static & dynamic, but the nature of OIDC is that it's typical new RPs get added via code, without any intervention. For example, adding a new app that uses "Login with Facebook" (or any other OIDC OP).

Reply:
> does it make sense to consolidate them and move to the C2IdOpTestCase parent class

I don't think so, because each test is responsible for its own client, and I don't think we want to encourage re-use there. It would just create too many dependencies between tests.